### PR TITLE
rbd: additional error checking in OpenImage, CreateImage and RemoveImage

### PR DESCRIPTION
**Pull request by nixpanic**
When passing `nil` arguments or empty strings for names some librbd functions may segfault. We do not like this, so we catch the errors early and return something to the called.

## Checklist
- [x] Added tests for features and functional changes
- [ ] ~Public functions and types are documented~
- [x] Standard formatting is applied to Go code

Reported-by: @madhu-1

### CODE REVIEW

File: rbd/rbd.go [hunk 1]
```diff
@@ -1101,6 +1115,13 @@ func OpenImageReadOnly(ioctx *rados.IOContext, name, snapName string) (*Image, e
 //  int rbd_open_by_id(rados_ioctx_t io, const char *id,
 //                     rbd_image_t *image, const char *snap_name);
 func OpenImageById(ioctx *rados.IOContext, id, snapName string) (*Image, error) {
+	if ioctx == nil {
```
Comment (Madhu-1) [inline rbd/rbd.go:1118]: can you avoid code duplication by creating a new helper/validator function?

**Comment by agarwal-mudit** [comment]
> LGTM. Do we want to add to this pr for recently merged functions or do it in a follow-up?

Added the checks in https://github.com/ceph/go-ceph/pull/301

File: rbd/rbd.go (hunk 1 above)
Comment (nixpanic) [reply rbd/rbd.go:1118]: Unfortunately I do not see how. We have `Image.validate()`, but we can not use it in this case where different parameters are passed.

Comment (phlogistonjohn) [reply rbd/rbd.go:1118]: It's unfortunate but I agree with @nixpanic there's not much else to do for these.

**Pull request merged**